icu4c/source/test/intltest/uts46test.cpp

```diff
@@ -245,6 +247,20 @@ void UTS46Test::TestNotSTD3() {
     }
 }
 
+void UTS46Test::TestNonASCII() {
+    IcuTestErrorCode errorCode(*this, "TestNonASCII()");
+    LocalPointer<IDNA> idna(IDNA::createUTS46Instance(0, errorCode));
+    if(errorCode.isFailure()) {
+        return;
+    }
+    UnicodeString input(u"xn--pleѐ");  // ends with non-ASCII U+0450
+    UnicodeString result;
+    IDNAInfo info;
+    idna->nameToUnicode(input, result, info, errorCode);
+    assertTrue("nameToUnicode() should detect non-ASCII",
+               (info.getErrors()&UIDNA_ERROR_PUNYCODE)!=0);
+}
```

Comment:
since some of the ASCII which would caused basicToDigit[ch] to get -1 will now get value < -1, could we add some test for those chars? such as "]", "[" and ">" "<". I know from the current code it is only used in one place right after that but I think adding some test of that which assert it will fail will be nice. Something like u"xn--ple[" and u"xn--p<le"

Reply:
Good idea, done. Will force-push an amended commit.